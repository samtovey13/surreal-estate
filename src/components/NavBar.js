import React from "react";
import '../styles/NavBar.css';
import logo from '../images/logo.png';
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebook } from "@fortawesome/free-brands-svg-icons";
import FacebookLogin from "react-facebook-login/dist/facebook-login-render-props";

const NavBar = ({ onLogin, userId, onLogout }) => {
  return (
    <div className="navbar">
      <div className="header-flex-container">
        <div className="header">
          <img className="logo" src={logo} alt="logo" width="40" height="50" />
          <h1 className="header-name">Surreal Estate</h1>
        </div>
        {userId && (
          <button className="logout-button" onClick={onLogout}>
            <FontAwesomeIcon
              icon={faFacebook}
              className="facebook-logout-icon"
            />
            <p className="facebook-button-text">Sign Out</p>
          </button>
        )}
        {!userId && (
          <FacebookLogin
            appId="298673424934631"
            autoLoad={false}
            fields="name,email,picture"
            callback={onLogin}
            render={(renderProps) => (
              <button className="login-button" onClick={renderProps.onClick}>
                <FontAwesomeIcon
                  icon={faFacebook}
                  className="facebook-login-icon"
                />
                <p className="facebook-button-text">Login</p>
              </button>
            )}
          />
        )}
      </div>
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/">View Properties</Link>
        </li>
        {userId && (
          <li className="navbar-links-item">
            <Link to="/favourites">My Favourites</Link>
          </li>
        )}
        <li className="navbar-links-item">
          <Link to="/add-property">Add a Property</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
