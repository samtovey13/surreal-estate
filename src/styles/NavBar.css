.navbar {
  background-color: thistle;
}
.header-name {
  margin: 0;
  display: inline-block;
}

.navbar-links {
  margin: 0;
  padding: 0;
}

.navbar-links-item {
  margin: 0;
  padding: 10px 20px;
  display: inline-block;
}