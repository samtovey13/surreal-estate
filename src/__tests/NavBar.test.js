import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

import FacebookLogin from "react-facebook-login/dist/facebook-login-render-props";
jest.mock("react-facebook-login/dist/facebook-login-render-props");

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
