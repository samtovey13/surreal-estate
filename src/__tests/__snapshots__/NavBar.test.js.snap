// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="header-flex-container"
    >
      <div
        class="header"
      >
        <img
          alt="logo"
          class="logo"
          height="50"
          src="logo.png"
          width="40"
        />
        <h1
          class="header-name"
        >
          Surreal Estate
        </h1>
      </div>
    </div>
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
